services/gundeck/src/Gundeck/ThreadBudget.hs

```diff
@@ -23,10 +23,14 @@ module Gundeck.ThreadBudget
   , runWithBudget'
   , watchThreadBudgetState
 
-  -- * for testing
+  -- * for testing: FUTUREWORK: factor out into a library?
```

Comment:
I suggest we move this module to `Gundeck.ThreadBudget.Internal`, and make `Gundeck.ThreadBudget` a very simple module that only re-exports what we need for non-testing purposes.  Perhaps you can do that in #892?